framework/src/modules/chain/block_processor_v0.js

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright © 2019 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+'use strict';
+
+const BigNum = require('@liskhq/bignum');
+const {
+	BIG_ENDIAN,
+	hash,
+	hexToBuffer,
+	intToBuffer,
+	LITTLE_ENDIAN,
+	signDataWithPrivateKey,
+} = require('@liskhq/lisk-cryptography');
+const { validator } = require('@liskhq/lisk-validator');
+const { BlockProcessor } = require('./processor');
+const { baseBlockSchema } = require('./blocks');
+
+const SIZE_INT32 = 4;
+const SIZE_INT64 = 8;
+
+const getBytes = block => {
+	const blockVersionBuffer = intToBuffer(
+		block.version,
+		SIZE_INT32,
+		LITTLE_ENDIAN,
+	);
+
+	const timestampBuffer = intToBuffer(
+		block.timestamp,
+		SIZE_INT32,
+		LITTLE_ENDIAN,
+	);
+
+	const previousBlockBuffer = block.previousBlock
+		? intToBuffer(block.previousBlock, SIZE_INT64, BIG_ENDIAN)
+		: Buffer.alloc(SIZE_INT64);
+
+	const numTransactionsBuffer = intToBuffer(
+		block.numberOfTransactions,
+		SIZE_INT32,
+		LITTLE_ENDIAN,
+	);
+
+	const totalAmountBuffer = intToBuffer(
+		block.totalAmount.toString(),
+		SIZE_INT64,
+		LITTLE_ENDIAN,
+	);
+
+	const totalFeeBuffer = intToBuffer(
+		block.totalFee.toString(),
+		SIZE_INT64,
+		LITTLE_ENDIAN,
+	);
+
+	const rewardBuffer = intToBuffer(
+		block.reward.toString(),
+		SIZE_INT64,
+		LITTLE_ENDIAN,
+	);
+
+	const payloadLengthBuffer = intToBuffer(
+		block.payloadLength,
+		SIZE_INT32,
+		LITTLE_ENDIAN,
+	);
+
+	const payloadHashBuffer = hexToBuffer(block.payloadHash);
+
+	const generatorPublicKeyBuffer = hexToBuffer(block.generatorPublicKey);
+
+	const blockSignatureBuffer = block.blockSignature
+		? hexToBuffer(block.blockSignature)
+		: Buffer.alloc(0);
+
+	return Buffer.concat([
+		blockVersionBuffer,
+		timestampBuffer,
+		previousBlockBuffer,
+		numTransactionsBuffer,
+		totalAmountBuffer,
+		totalFeeBuffer,
+		rewardBuffer,
+		payloadLengthBuffer,
+		payloadHashBuffer,
+		generatorPublicKeyBuffer,
+		blockSignatureBuffer,
+	]);
+};
+
+const validateSchema = ({ block }) => {
+	const errors = validator.validate(baseBlockSchema, block);
+	if (errors.length) {
+		throw errors;
+	}
+};
+
+class BlockProcessorV0 extends BlockProcessor {
+	constructor({ blocksModule, logger, constants, exceptions }) {
+		super();
+		this.blocksModule = blocksModule;
+		this.logger = logger;
+		this.constants = constants;
+		this.exceptions = exceptions;
+
+		this.init.pipe([() => this.blocksModule.init()]);
```

Comment:
`this.blocksModule.init()` is added into both versions of block processor. Can't it be moved to base_block_processor?

Reply:
it could, but we didn't expose dynamically removing the stage from pipe. so if someone wants to fully custamize, it wouldn't be possible.
That's why I tried to keep the block_processor minimal